### PR TITLE
Add pallet-xcm::transfer_assets_using_type_and_then() for complex asset transfers

### LINKED ISSUE #281: Sending bridged DOT from AH to parachain fails after upgrade
Before the last upgrade at block [6793666](https://assethub-kusama.subscan.io/extrinsic/6793666-2) there's an example of a *limited reserve transfer assets* of DOT from Kusama asset hub to parachain `2281` which was successful, after the upgrade we can see another example at block [6814790](https://assethub-kusama.subscan.io/extrinsic/6814790-2) which fails with the error `InvalidAssetUnknownReserve`. It seems like a regression unless sending such foreign asset to a Kusama parachain is expected to be done differently now.

### PR DESCRIPTION
**Pull request by acatangiu**
Bump `pallet-xcm` to 8.0.5 to include `transfer_assets_using_type_and_then()` call to facilitate moving bridged assets around the ecosystem.

Example [test for bridged asset AssetHub->Parachain](https://github.com/paritytech/polkadot-sdk/blob/e0584a153df63ff138d12764085422ed06de548a/cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/hybrid_transfers.rs#L150C1-L156C54) - the actual call used [here](https://github.com/paritytech/polkadot-sdk/blob/e0584a153df63ff138d12764085422ed06de548a/cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/hybrid_transfers.rs#L57C1-L70C3).
Example [test for bridged asset Parachain->AssetHub](https://github.com/paritytech/polkadot-sdk/blob/e0584a153df63ff138d12764085422ed06de548a/cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/hybrid_transfers.rs#L278C1-L285C54) - the actual call used [here](https://github.com/paritytech/polkadot-sdk/blob/e0584a153df63ff138d12764085422ed06de548a/cumulus/parachains/integration-tests/emulated/tests/assets/asset-hub-westend/src/tests/hybrid_transfers.rs#L76C2-L89C3).

More details on the backported call available here: https://github.com/paritytech/polkadot-sdk/pull/4462

Fixes #281

### CODE REVIEW

**Review by joepetrowski — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 - Add `pallet-vesting` to Asset Hubs ([polkadot-fellows/runtimes#269](https://github.com/polkadot-fellows/runtimes/pull/269))
 - Add Pay Salary Collectives test ([polkadot-fellows/runtimes#260](https://github.com/polkadot-fellows/runtimes/pull/260))
+- Add pallet-xcm::transfer_assets_using_type_and_then() for complex asset transfers ([polkadot-fellows/runtimes#311](https://github.com/polkadot-fellows/runtimes/pull/311))
```
Comment (bkontur) [inline CHANGELOG.md:18]:
```suggestion
- Add `pallet-xcm::transfer_assets_using_type_and_then()` for complex asset transfers ([polkadot-fellows/runtimes#311](https://github.com/polkadot-fellows/runtimes/pull/311))
```

**Review by bkontur — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

**Review by serban300 — APPROVED** [review]

**Comment by acatangiu** [comment]
cc @muharem, please need another rank2 review here

**Review by xlc — APPROVED** [review]

**Pull request merged**